### PR TITLE
Initial commit

### PR DESCRIPTION
**Pull request by monzonj**

### CODE REVIEW

File: build.sbt [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+    organization := "com.leonteq",
+    exportJars := true,
+    version := "0.0.1",
+    parallelExecution in test := false,
+    fork in run := true,
+    publishArtifact in (Compile, packageDoc) := false,
+    publishArtifact in (Test, packageBin) := true,
+    concurrentRestrictions in Global += Tags.limit(Tags.Test, 1)
+  )
+  .settings(Settings.settings: _*)
```
Comment (Reeebuuk) [inline build.sbt]: duplicate from above

File: project/Dependencies.scala [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+  // to use in build.sbt
+  lazy val runtime: Seq[ModuleID] = akka ++ playJson ++ pureConfig ++ logging  ++ oidc
+  lazy val test: Seq[ModuleID] = akkaTest ++ scalaTest
+
+  // Prod dependencies
+  private lazy val akka: Seq[ModuleID] = Seq(
+    "com.typesafe.akka" %% "akka-contrib"           % akkaVersion,
+    "com.typesafe.akka" %% "akka-stream"            % akkaVersion,
+    "com.typesafe.akka" %% "akka-cluster"           % akkaVersion,
+    "com.typesafe.akka" %% "akka-cluster-sharding"  % akkaVersion,
```
Comment (Reeebuuk) [inline project/Dependencies.scala]: cluster sharding? cluster? cors?

File: src/main/scala/com/leonteq/eportal/oidc/OidcProtocol.scala [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+import java.net.URLEncoder
+import java.nio.charset.StandardCharsets
+import java.util.{ Base64, UUID }
+
+import akka.http.scaladsl.server.Rejection
+import play.api.libs.json._
+
+object OidcProtocol {
+
+  case class UserId(value:        String) extends AnyVal
```
Comment (Reeebuuk) [inline src/main/scala/com/leonteq/eportal/oidc/OidcProtocol.scala]: Would rather remove UserId and CorrelationId, will run deep into the app

File: src/main/scala/com/leonteq/eportal/oidc/logging/ApplicationLogger.scala [hunk 4]
```diff
@@ -0,0 +1,50 @@
+package com.leonteq.eportal.oidc.logging
+
+import java.io.{ ByteArrayInputStream, ByteArrayOutputStream, ObjectInputStream, ObjectOutputStream }
+
+import com.leonteq.eportal.oidc.OidcProtocol.CorrelationId
+import com.typesafe.scalalogging.{ CanLog, Logger, LoggerTakingImplicit }
+import org.slf4j.MDC
+
+trait ApplicationLogger {
```
Comment (Reeebuuk) [inline src/main/scala/com/leonteq/eportal/oidc/logging/ApplicationLogger.scala]: would remove this as well

File: src/main/scala/com/leonteq/eportal/oidc/logging/CorrelationIdConverter.scala [hunk 5]
```diff
@@ -0,0 +1,13 @@
+package com.leonteq.eportal.oidc.logging
+
+import ch.qos.logback.classic.pattern.ClassicConverter
+import ch.qos.logback.classic.spi.ILoggingEvent
+import org.slf4j.MDC
+
+class CorrelationIdConverter extends ClassicConverter {
```
Comment (Reeebuuk) [inline src/main/scala/com/leonteq/eportal/oidc/logging/CorrelationIdConverter.scala]: as well

File: src/test/resources/logback.xml [hunk 6]
```diff
@@ -0,0 +1,63 @@
...
+
+    <!-- Console appender, not to use it in deployed environments -->
+    <appender name="stdout" class="ch.qos.logback.core.ConsoleAppender">
+        <encoder>
+            <pattern>${LAYOUT}</pattern>
+        </encoder>
+    </appender>
+
+    <!-- Main application log file -->
+    <appender name="logs" class="ch.qos.logback.core.rolling.RollingFileAppender">
```
Comment (Reeebuuk) [inline src/test/resources/logback.xml]: missing the structure for ES

File: src/test/resources/logback.xml [hunk 7]
```diff
@@ -0,0 +1,63 @@
...
+        </rollingPolicy>
+    </appender>
+
+
+    <appender name="logsAsync" class="ch.qos.logback.classic.AsyncAppender">
+        <appender-ref ref="logs" />
+    </appender>
+
+    <root level="INFO">
+        <appender-ref ref="stdout"/> <!-- remove this for prd -->
```
Comment (Reeebuuk) [inline src/test/resources/logback.xml]: ?

File: build.sbt (hunk 1 above)
Comment (monzonj) [reply build.sbt]: thanks

File: project/Dependencies.scala (hunk 2 above)
Comment (monzonj) [reply project/Dependencies.scala]: all gone already

File: src/main/scala/com/leonteq/eportal/oidc/OidcProtocol.scala (hunk 3 above)
Comment (monzonj) [reply src/main/scala/com/leonteq/eportal/oidc/OidcProtocol.scala]: I renamed CorrelationId to OidcCorrelationId. Kept UserId. Tested and works fine. 
Should I get rid of it?

File: src/main/scala/com/leonteq/eportal/oidc/logging/ApplicationLogger.scala (hunk 4 above)
Comment (monzonj) [reply src/main/scala/com/leonteq/eportal/oidc/logging/ApplicationLogger.scala]: ok. will do

File: src/test/resources/logback.xml (hunk 6 above)
Comment (monzonj) [reply src/test/resources/logback.xml]: You mean your logging stuff?

Comment (Reeebuuk) [reply src/test/resources/logback.xml]: yes

File: src/main/scala/com/leonteq/eportal/oidc/OidcProtocol.scala (hunk 3 above)
Comment (Reeebuuk) [reply src/main/scala/com/leonteq/eportal/oidc/OidcProtocol.scala]: Problem is that we'll have UserId from our app and UserId from here and who knows what will lcm have. Better to leave it to the apps. Cuz UserId is just a wrapper, not really a part of OIDC lib

File: src/test/resources/logback.xml (hunk 6 above)
Comment (monzonj) [reply src/test/resources/logback.xml]: I'd rather have that in another story. For now I'll use basic logging, so squad1 can easily use it.

**Pull request merged**